ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7759,6 +7759,18 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
               .getMsg(destinationPath.toUri().toString()));
         }
       }
+      // handle direct insert CTAS case
+      // for direct insert CTAS, the table creation DDL is not added to the task plan in TaskCompiler,
+      // therefore we need to add the InsertHook here manually so that HiveMetaHook#commitInsertTable is called
```

Comment:
Where do we add the InsertHook here? I do not really grok the comment.

Reply:
it's added inside the `createPreInsertDesc` method. I'll rework the comment to make it easier to follow